code/script.js

```diff
@@ -1,51 +1,182 @@
+          </div>
+        </section>
+      `;
+    }
+    chat.scrollTop = chat.scrollHeight;
+  }
+
+  // This function will get the bot to progress the conversation
+  const generateRequest = (message) => {
+    console.log('questionNumber', questionNumber);
+    
+    if (questionNumber === 1) {
+      setTimeout (() => showOptions(message), 1000);
+    } else if (questionNumber === 2) {
+      setTimeout (() => passengers(message), 1000);
+    } else if (questionNumber === 3) {
+      setTimeout (() => addBaggage(message), 1000);
+    } else if (questionNumber === 4) {
+      setTimeout (() => confirmBooking(message), 1000);
+    } else if (questionNumber === 5) {
+      setTimeout (() => goodbye(message), 1000);
+    }
+  }
+
+  // Greeting from the bot turns up when the page is loaded
+  const greeting = () => {
+    questionNumber = 1;
+    botReply('Hello there. What is your name?');
+  }
+
+  // The function that makes the users name-response turn up
+  const handleInput = (event) => {
+      event.preventDefault();
+      const reply = userInput.value;
+      userReply(reply);
+      userInput.value = '';
+
+      if (reply !== '') {
+        setTimeout (() => showOptions(reply), 1000);
+      } else {
+        setTimeout (() => botReply('That is not a valid answer'), 300); 
+      }
+
+  }
+
+  // the bots first question; a multiple choise question with three buttons with different alternatives
+  const showOptions = (message) => {
+    questionNumber++;
+    botReply(`Hi ${message}, How would you like to travel?`);
+
+      inputWrapper.innerHTML = `
+        <button id='boatBtn'>Boat</button>
+        <button id='busBtn'>Bus</button>
+        <button id='trainBtn'>Train</button>
+      `;
+
+      document.getElementById('boatBtn') .addEventListener('click', () => {
+        generateRequest('Boat');
+        userReply('Boat');
+      });
+      document.getElementById('busBtn') .addEventListener('click', () => {
+        generateRequest('Bus');
+        userReply('Bus');
+      });
+      document.getElementById('trainBtn') .addEventListener('click', () => {
+        generateRequest('Train');
+        userReply('Train');
+      });
+  }
+
+  // the bots second question; a list of numbers to choose from
+  const passengers = (transport) => {
+    questionNumber++;
+    botReply(`${transport} you say, how fun! How many passengers will you be?`);
+      
+    inputWrapper.innerHTML =`
+      <select id='numberOfPassengers' onchange='changeFunc'>
+        <option value='0'>0</option>
+        <option value='1'>1</option>
+        <option value='2'>2</option>
+        <option value='3'>3</option>
+      </select>
+    `;
+
+    document.getElementById('numberOfPassengers') .onchange = () => {
+      generateRequest(numberOfPassengers.value);
+      userReply(numberOfPassengers.value);
+    }
+  }
+
+  // the bots third question; a multiple choise question with 'yes'- and 'no'-button
+  const addBaggage = () => {
```

Comment:
Maybe ask how much luggage I want to transport before I confirm (or not confirm) the booking?